addons/web/static/src/js/widgets/smooth_scroll.js

```diff
@@ -0,0 +1,315 @@
+odoo.define('web.smooth_scroll_on_drag', function (require) {
+"use strict";
+
+const Class = require('web.Class');
+const Mixins = require('web.mixins');
+
+/**
+ * Provides a helper for SmoothScrollOnDrag options.offsetElements
+ */
+const OffsetElementsHelper = Class.extend({
```

Comment:
As discussed in discord, I think there is a problem with this. I don't know about the logic but while testing there are new problems that are not there in master.